pkg/storage/fs/owncloud/upload.go

```diff
@@ -375,6 +383,54 @@ func (upload *fileUpload) FinishUpload(ctx context.Context) error {
 			}
 		}
 	*/
+	log := appctx.GetLogger(upload.ctx)
+
+	sha1h := make([]byte, 0, 32)
+	md5h := make([]byte, 0, 32)
+	adler32h := make([]byte, 0, 32)
+	{
+		sh := sha1.New()
+		mh := md5.New()
+		ah := adler32.New()
+		f, err := os.Open(upload.binPath)
+		if err != nil {
+			log.Err(err).Msg("Decomposedfs: could not open file for checksumming")
+			// we can continue if no oc checksum header is set
+		}
+		defer f.Close()
+
+		r1 := io.TeeReader(f, sh)
+		r2 := io.TeeReader(r1, mh)
```

Comment:
Ok, bear with me, but.... short variable names have a long history of proving that they're not cool. We tried that in C already back in the dark days, paying back with lots of head scratching... So why not just spend a few more chars to get speaking names? Yes, also for dumb, short living local variables... Ok, I am not blaming you here for the golang way of doing things ;-)